jmh-benchmarks/src/main/java/org/apache/kafka/jmh/assignor/ServerSideAssignorBenchmark.java

```diff
@@ -0,0 +1,266 @@
+
+            // Define buckets for each member and assign topics from the same bucket
+            for (int bucket = 0; bucket < bucketCount; bucket++) {
+                int memberStartIndex = bucket * bucketSizeMembers;
+                int memberEndIndex = Math.min((bucket + 1) * bucketSizeMembers, numberOfMembers);
+
+                int topicStartIndex = bucket * bucketSizeTopics;
+                int topicEndIndex = Math.min((bucket + 1) * bucketSizeTopics, topicCount);
+
+                Set<Uuid> bucketTopics = new HashSet<>(allTopicIds.subList(topicStartIndex, topicEndIndex));
+
+                // Assign topics to each member in the current bucket
+                for (int i = memberStartIndex; i < memberEndIndex; i++) {
+                    addMemberSpec(members, i, new HashSet<>(bucketTopics));
+                }
+            }
+        }
+
+        this.assignmentSpec = new AssignmentSpec(members);
+    }
+
+    private Optional<String> rackId(int index) {
+        return isRackAware ? Optional.of("rack" + index % NUMBER_OF_RACKS) : Optional.empty();
+    }
+
+    private void addMemberSpec(
+        Map<String, AssignmentMemberSpec> members,
+        int memberIndex,
+        Set<Uuid> subscribedTopicIds
+    ) {
+        String memberId = "member" + memberIndex;
+        Optional<String> rackId = rackId(memberIndex);
+
+        members.put(memberId, new AssignmentMemberSpec(
+            Optional.empty(),
+            rackId,
+            subscribedTopicIds,
+            Collections.emptyMap()
+        ));
+    }
+
+    private static Map<Integer, Set<String>> mkMapOfPartitionRacks(int numPartitions) {
+        Map<Integer, Set<String>> partitionRacks = new HashMap<>(numPartitions);
+        for (int i = 0; i < numPartitions; i++) {
+            partitionRacks.put(i, new HashSet<>(Arrays.asList("rack" + i % NUMBER_OF_RACKS, "rack" + (i + 1) % NUMBER_OF_RACKS)));
+        }
+        return partitionRacks;
+    }
+
+    private void simulateIncrementalRebalance(Map<Uuid, TopicMetadata> topicMetadata) {
+        GroupAssignment initialAssignment = partitionAssignor.assign(assignmentSpec, subscribedTopicDescriber);
+        Map<String, MemberAssignment> members = initialAssignment.members();
+
+        Map<String, AssignmentMemberSpec> updatedMembers = new HashMap<>();
+        members.forEach((memberId, memberAssignment) -> {
+            AssignmentMemberSpec memberSpec = assignmentSpec.members().get(memberId);
+            updatedMembers.put(memberId, new AssignmentMemberSpec(
+                memberSpec.instanceId(),
+                memberSpec.rackId(),
+                memberSpec.subscribedTopicIds(),
+                memberAssignment.targetPartitions()
+            ));
+        });
+
+        Optional<String> rackId = rackId(memberCount - 1);
+        updatedMembers.put("newMember", new AssignmentMemberSpec(
+            Optional.empty(),
+            rackId,
+            topicMetadata.keySet(),
```

Comment:
This would be common for both homogeneous and heterogenous and honestly I think it makes sense to test when a member could potentially affect all the topics. It is possible to add a member that is subscribed to all the topics right

Reply:
I think that we could test this too but it is a little unexpected in this context. We basically said that we want to bucket members into 5 groups in the heterogenous case and we don't respect it here. Don't you think that it would be better to respect it? We could add another group topology in the future if we want to.